src/jinja2/environment.py

```diff
@@ -811,16 +812,11 @@ def _load_template(self, name, globals):
             if template is not None and (
                 not self.auto_reload or template.is_up_to_date
             ):
-                # update globals if changed
-                new_globals = globals.items() - template.globals.items()
-                if new_globals:
-                    # it is possible for the template and environment to share
-                    # a globals object, in which case, a new copy should be
-                    # made to avoid affecting other templates
-                    if template.globals is self.globals:
-                        template.globals = dict(template.globals, **globals)
-                    else:
-                        template.globals.update(dict(new_globals))
+                # template.globals is a ChainMap, modifying it will only
+                # affect the template, not the environment globals.
+                if globals:
+                    template.globals.update(globals)
+
```

Comment:
The if-statement will always be true since the `ChainMap` is always instantiated with env globals (unless a user empties out the default env globals). 

I think this will only be an issue when the template and environment globals overlap. (I didn't see the docs explicitly mention intersections, but I assume it's possible as long as template globals have higher precedence.) So if allowed, something like this fails:

```python
    def test_globals_overlap(self):
        env = Environment(
            loader=DictLoader({"base": "{{ foo }}", "main": "{% include 'base' %}"})
        )

        env.globals['foo'] = 'env'
        tmpl = env.get_template("main", globals={"foo": 1})
        assert tmpl.render() == "1"

        tmpl = env.get_template("base", globals={"foo": 2})
        assert tmpl.render() == "2"

        tmpl = env.get_template("main")
        assert tmpl.render() == "1" # got env instead

        tmpl = env.get_template("base")
        assert tmpl.render() == "2" # got env instead

```

Reply:
Those results seem correct based on your original implementation, where template globals are additive. Loading without globals just means the template's globals aren't changed further. Good point about the if statement though, I'll check the logic in general.

Reply:
I think I misread your example, it's saying you're not seeing the template globals anymore, so yeah I'll fix that.

Reply:
Changed it so `_load_template` calls `make_globals` instead of `get_template` and `select_template`. Updated the test to demonstrate that.